go/protocol/keybase1/extras.go

```diff
@@ -1518,23 +1518,48 @@ func (t TeamName) IsNil() bool {
 
 // underscores allowed, just not first or doubled
 var namePartRxx = regexp.MustCompile(`([a-zA-Z0-9][a-zA-Z0-9_]?)+`)
+var implicitRxxString = fmt.Sprintf("^%s[0-9a-f]{%d}$", implicitTeamPrefix, implicitSuffixLengthBytes*2)
+var implicitNameRxx = regexp.MustCompile(implicitRxxString)
+
+var implicitTeamPrefix = "__keybase_implicit_team__"
+var implicitSuffixLengthBytes = 16
+
+func TeamNameFromString(s string) (TeamName, error) {
+	ret := TeamName{}
+	var regularErr, implicitErr error
 
-func TeamNameFromString(s string) (ret TeamName, err error) {
 	parts := strings.Split(s, ".")
 	if len(parts) == 0 {
 		return ret, errors.New("need >= 1 part, got 0")
 	}
+
+	if len(parts) != 1 || !implicitNameRxx.MatchString(s) {
```

Comment:
Right, that'd work as long as there's only one way for it to fail (which is true for now, but maybe later we'd like to give better error messages?). But I can just simplify it now if you like since it makes it easier to read.